src/Integration.Vsix/ErrorList/SnapshotFactory.cs

```diff
@@ -18,17 +18,29 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.Linq;
 using Microsoft.VisualStudio.Shell.TableManager;
+using SonarLint.VisualStudio.Core.Helpers;
 
 namespace SonarLint.VisualStudio.Integration.Vsix
 {
+    internal interface ISnapshotFactory : ITableEntriesSnapshotFactory
```

Comment:
`Snapshot` is used both the the error list and text span code. It would be better to be more explicit about which one we are talking about here e.g. `IIssuesSnaphotFactory`.

It might be worth considering making the namespace more specific too.